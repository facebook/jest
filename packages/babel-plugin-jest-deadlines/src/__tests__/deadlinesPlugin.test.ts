/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/* eslint-disable sort-keys */

import pluginTester from 'babel-plugin-tester';
import babelPluginJestDeadlines from '..';

pluginTester({
  plugin: babelPluginJestDeadlines,
  pluginName: 'babel-plugin-jest-deadlines',
  tests: {
    'bare function call in statement context': {
      // language=JavaScript
      code: `
        async function foo() {
          await bar();
        }`,
      snapshot: true,
    },
    'transforms an expression': {
      // language=JavaScript
      code: `
        async function foo() {
          return await bar();
        }`,
      snapshot: true,
    },
    'transforms an existing expect': {
      // language=JavaScript
      code: `
        async function foo() {
          await expect(bar()).resolves.toBe("hot potatoes");
        }`,
      snapshot: true,
    },
    'skips double stacking': {
      // language=JavaScript
      code: `
        async function foo() {
          await expect.withinDeadline(bar());
        }`,
      snapshot: true,
    },
    'transforms multiple awaits': {
      // language=JavaScript
      code: `
        async function foo() {
          await bar(1, 2, 3);
          await quux(1, 2, 3);
        }`,
      snapshot: true,
    },
    'transforms await in argument context': {
      // language=JavaScript
      code: `
        async function foo() {
          await bar(1, await quux(), 3);
        }`,
      snapshot: true,
    },
  },
});
