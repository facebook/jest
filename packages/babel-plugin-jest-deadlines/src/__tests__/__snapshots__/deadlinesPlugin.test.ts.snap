// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-jest-deadlines bare function call in statement context: bare function call in statement context 1`] = `

async function foo() {
  await bar();
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  await _expect.withinDeadline(bar());
}


`;

exports[`babel-plugin-jest-deadlines skips double stacking: skips double stacking 1`] = `

async function foo() {
  await expect.withinDeadline(bar());
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  await _expect.withinDeadline(expect.withinDeadline(bar()));
}


`;

exports[`babel-plugin-jest-deadlines transforms an existing expect: transforms an existing expect 1`] = `

async function foo() {
  await expect(bar()).resolves.toBe("hot potatoes");
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  await _expect.withinDeadline(expect(bar()).resolves.toBe("hot potatoes"));
}


`;

exports[`babel-plugin-jest-deadlines transforms an expression: transforms an expression 1`] = `

async function foo() {
  return await bar();
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  return await _expect.withinDeadline(bar());
}


`;

exports[`babel-plugin-jest-deadlines transforms await in argument context: transforms await in argument context 1`] = `

async function foo() {
  await bar(1, await quux(), 3);
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  await _expect.withinDeadline(bar(1, await _expect.withinDeadline(quux()), 3));
}


`;

exports[`babel-plugin-jest-deadlines transforms multiple awaits: transforms multiple awaits 1`] = `

async function foo() {
  await bar(1, 2, 3);
  await quux(1, 2, 3);
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _expect = require("@jest/globals").expect;

async function foo() {
  await _expect.withinDeadline(bar(1, 2, 3));
  await _expect.withinDeadline(quux(1, 2, 3));
}


`;
